vega_query/visualisations/overlay.py

```diff
@@ -894,6 +893,50 @@ def overlay_volume(
         y.append(price * size if price != 0 else np.nan)
     ax.step(x, y, label="volume", where="post")
 
+def overlay_cumulative_volume(
+    ax: Axes,
+    trades: List[protos.vega.vega.Trade],
+    price_decimals: int,
+    size_decimals: int,
+):
+    x = []  # List to store timestamps
+    y = []  # List to store cumulative volume
+    cumulative_volume = 0  # Initialize cumulative volume
+
+    trades = reversed(trades)
+
+    for trade in trades:
+        # Convert timestamp to datetime for x-axis
+        timestamp = timestamp_to_datetime(trade.timestamp, nano=True)
+        x.append(timestamp)
+        
+        # Ensure price and size are available
+        if hasattr(trade, 'price') and hasattr(trade, 'size'):
+            price = padded_int_to_float(trade.price, price_decimals)
+            size = padded_int_to_float(trade.size, size_decimals)
+            
+            # Calculate traded volume (price * size)
+            volume = price * size if price != 0 else 0
+            
+            # Debugging: Print current values
+            print(f"Trade timestamp: {trade.timestamp}, Price: {price}, Size: {size}, Volume: {volume}")
+            
+            # Accumulate volume (ensure volume is positive)
+            cumulative_volume += abs(volume)
+            y.append(cumulative_volume)
+        else:
+            # If price or size is missing, assume no change in cumulative volume
+            y.append(cumulative_volume)
+            print(f"Missing price or size for trade at {trade.timestamp}")
+
+    # Plot the cumulative volume data using step plot
+    ax.step(x, y, label="Cumulative Volume", where="post")
```

Comment:
Related to above, can unpack the kwargs here.
```suggestion
    ax.step(x, y, label="Cumulative Volume", where="post", **kwargs)
```